src/random_lcg.C

```diff
@@ -0,0 +1,142 @@
+#include "random_lcg.h"
+#include <cmath>
+
+
+// Starting seed
+int64_t seed = 1;
+
+// LCG parameters
+const uint64_t prn_mult   = 2806196910506780709LL;   // multiplication factor, g
+const uint64_t prn_add    = 1;                       // additive factor, c
+const uint64_t prn_mod    = 0x8000000000000000;      // 2^63
+const uint64_t prn_mask   = 0x7fffffffffffffff;      // 2^63 - 1
+const uint64_t prn_stride = 152917LL;                // stride between particles
+const double   prn_norm   = pow(2, -63);             // 2^-63
+
+// Module constants
+const int N_STREAMS         = 5;
+const int STREAM_TRACKING   = 0;
+const int STREAM_TALLIES    = 1;
+const int STREAM_SOURCE     = 2;
+const int STREAM_URR_PTABLE = 3;
+const int STREAM_VOLUME     = 4;
+
+// Current PRNG state
+uint64_t prn_seed[N_STREAMS];  // current seed
+int      stream;               // current RNG stream
+#pragma omp threadprivate(prn_seed, stream)
+
+
+//==============================================================================
+// PRN
+//==============================================================================
+
+extern "C" double
+prn()
+{
+  // This algorithm uses bit-masking to find the next integer(8) value to be
+  // used to calculate the random number.
+  prn_seed[stream] = (prn_mult*prn_seed[stream] + prn_add) & prn_mask;
+
+  // Once the integer is calculated, we just need to divide by 2**m,
+  // represented here as multiplying by a pre-calculated factor
+  return prn_seed[stream] * prn_norm;
+}
+
+//==============================================================================
+// FUTURE_PRN
+//==============================================================================
+
+extern "C" double
```

Comment:
How attached are you to this suggestion?  I personally try to avoid massive bracketed blocks where possible.  Plus, I like being able to find all the modifiers to a function/variable in one place.

Reply:
I'm not attached to anything at this point since we haven't adopted a style guide for C++.